tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -0,0 +1,108 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/s2n_tls.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+    
+    /* s2n_tls13_server_nst_send */
+    {
+        /* Check session ticket message is correctly written. The contents of the 
+         * message will be tested more thoroughly once the s2n_tls13_server_nst_recv 
+         * function is written. */
+        {
+            /* Session ticket keys. Taken from test vectors in https://tools.ietf.org/html/rfc5869 */
+            uint8_t ticket_key_name[16] = "2016.07.26.15\0";
+            S2N_BLOB_FROM_HEX(ticket_key, 
+            "077709362c2e32df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5");
+
+            S2N_BLOB_FROM_HEX(test_master_secret,
+            "ee85dd54781bd4d8a100589a9fe6ac9a3797b811e977f549cd"
+            "531be2441d7c63e2b9729d145c11d84af35957727565a4");
+
+            struct s2n_config *config;
+            uint64_t current_time;
+            struct s2n_connection *conn;
+            EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_NOT_NULL(config = s2n_config_new());
+            conn->actual_protocol_version = S2N_TLS13;
+            conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
+            uint16_t test_tickets_sent = 10;
+            conn->tickets_sent = test_tickets_sent;
+
+            /* Set up encryption key */
+            EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(config, 1));
+            EXPECT_SUCCESS(config->wall_clock(config->sys_clock_ctx, &current_time));
+            EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(config, ticket_key_name, strlen((char *)ticket_key_name),
+                         ticket_key.data, sizeof(ticket_key), current_time/ONE_SEC_IN_NANOS));
+            
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+
+            /* Set up master resumption secret */
+            struct s2n_blob secret = { 0 };
+            struct s2n_stuffer secret_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&secret, conn->secure.master_secret, S2N_TLS_SECRET_LEN));
+            EXPECT_SUCCESS(s2n_stuffer_init(&secret_stuffer, &secret));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(&secret_stuffer, test_master_secret.data, S2N_TLS_SECRET_LEN));
+            
+            EXPECT_SUCCESS(s2n_tls13_server_nst_send(conn));
+
+            uint8_t message_type = 0;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint8(&conn->handshake.io, &message_type));
+            EXPECT_EQUAL(TLS_SERVER_NEW_SESSION_TICKET, message_type);
+
+            /* The TLS1.3 new session ticket message size cannot be tested here
+             * because the session resumption ticket secret varies from 28-48 bytes. */
+            EXPECT_SUCCESS(s2n_stuffer_skip_read(&conn->handshake.io, SIZEOF_UINT24));
```

Comment:
Not that I'm arguing that there's value in testing the size, but doesn't we know the size since we know the size of our test secret?

Reply:
We know the value of the master secret, but not the individual ticket secret.

Reply:
I would just check that the size matches the amount of data available in the output as a sanity check.